src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs

```diff
@@ -297,6 +297,58 @@ protected Size MeasureChild(View view, Size slotSize)
 				return ret;
 			}
 
+			if(frameworkElement != null && !(frameworkElement is FrameworkElement))
+			{
+				// For IFrameworkElement implementers that are not FrameworkElements, the constraint logic must
+				// be performed by the parent. Otherwise, the native element will take the size it needs without
+				// regards to explicit XAML size characteristics. The Android ProgressBar is a good example of
+				// that behavior.
```

Comment:
Are we able to extract and multi target this logic? At least it would be great to add a comment/link to and from where the code the was copied.

Reply:
There's no need for that at this point, this code will disappear once we add support for NativeElement.